agent/src/agent/pipeline/pipeline.py

```diff
@@ -250,9 +252,23 @@ def delay(self) -> str:
     def batch_size(self) -> str:
         return self.config.get('batch_size', 1000)
 
-    def set_streamsets(self, streamsets):
-        self.streamsets_id = streamsets.id
-        self.streamsets = streamsets
+    def get_config(self) -> dict:
```

Comment:
This name `get_config` might be confusing because the pipeline entity has a property `config`

Reply:
ok, I'll rename it here as well as in the IPipeline in the lib, just didn't want to give a long name to the interface method